### PR TITLE
Add Fido client and filetype attr

### PR DESCRIPTION
**Pull request by wtbarnes**
This pull request adds a new Fido client called `EISClient` that enables searches of the level 1 HDF5 and FITS files via Fido, the sunpy data retrieval interface. An additional attr is also provided to discriminate between the HDF data and header files and FITS files.

Though I've not deleted any code, I would recommend that this supersede the code that is in `download/download_hdf5_data.py`. 

A few things that could still be done:

- [x] Tests (there are doctests though I don't actually think these get run at the moment)
- [ ] ~~Remove `download/download_hdf5_data.py` as this provides duplicate functionality~~
- [x] Docs page for `net` subpackage

~~At the moment, this is difficult to test because eis.nrl.navy.mil is experiencing SSL certificate issues.~~

This PR should be squash merged.

### CODE REVIEW

File: eispac/net/attrs.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class FileType(SimpleAttr):
+    """
+    Specifies the type of EIS level 1 file
+    
+    Parameters
+    ----------
+    value: `str`
+        Either "data" or "head"
```
Comment (dstansby) [inline eispac/net/attrs.py:13]: I'd suggest being clearer and using "header" instead of "head"?

File: eispac/net/client.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    ----------------------- ----------------------- ---------- --------- ------ -------- ----- --------
+    2020-11-09 00:10:12.000 2020-11-09 00:10:12.999        EIS intensity HINODE      NRL     1     head
+    <BLANKLINE>
+    <BLANKLINE>
+    """
+    baseurl = r'https://eis.nrl.navy.mil/level1/hdf5/%Y/%m/%d/eis_%Y%m%d_%H%M%S.(\w){4}.h5'
+    pattern = '{}/hdf5/{year:4d}/{month:2d}/{day:2d}/eis_{:8d}_{hour:2d}{minute:2d}{second:2d}.{FileType}.{}'
+    
+    @property
+    def info_url(self):
```
Comment (dstansby) [inline eispac/net/client.py:58]: 🥳

**Comment by MJWeberg** [comment]
This is great! It looks like the data server issues are finally fixed now, so I will go ahead and merge this in and work on updating the docs to match.

For now, I think it is best to keep the `download_hdf5_data()` function, since it allows for a quick, direct download from the server (with basic handling of partial files) and, most importantly, it is currently used by the `eis_catalog` search and download GUI. However, this Fido client is a great tool for users who are looking for multi-spacecraft observations or just prefer to use a Python script instead.

One question for future consideration, is it possible to connect a database of metadata to a Fido client? There is a lot of coordinate and wavelength information in the catalog that I assume would be useful to include as search parametersin Fido (although the official EIS as-run catalog is a bit of a maze, so we might need to generate a reduced database for the just the HDF5 files).

**Comment by wtbarnes** [comment]
I think this is finally ready (only about a year late!)

Please squash this before merging. The commit history is a bit all over the place.

File: eispac/net/client.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    @classmethod
+    def _can_handle_query(cls, *query):
+        """
+        Check if this client can handle a given Fido query.
+        Returns
+        -------
+        bool
+            True if this client can handle the given query.
+        """
+        required = {a.Time, a.Instrument, a.Source}
```
Comment (nabobalis) [inline eispac/net/client.py:133]: Do you need to make Source required?

Comment (wtbarnes) [reply eispac/net/client.py:133]: There could be another future instrument called EIS? 🤷

**Comment by wtbarnes** [comment]
> One question for future consideration, is it possible to connect a database of metadata to a Fido client? There is a lot of coordinate and wavelength information in the catalog that I assume would be useful to include as search parameters in Fido (although the official EIS as-run catalog is a bit of a maze, so we might need to generate a reduced database for the just the HDF5 files).

@MJWeberg we talked about this a bit offline a few months ago, but just wanted to record my thoughts here for posterity. This is definitely possible. The difficulty lies in the fact that the EIS catalogue is not a *hosted* SQL database, but rather a local file that one needs to download. This means it's not just a matter of translating a Fido query into a query against the EIS catalogue, but also downloading the catalogue and making sure it is up to date, with the assumption that there is always an existing correspondence between this local catalogue and what is being provided on the remote server.

I think this is an interesting direction and would enable much more detailed queries, but is more complicated than what is proposed here. I would recommend spinning this out as a separate issue.

File: eispac/net/client.py (hunk 3 above)
Comment (Cadair) [reply eispac/net/client.py:135]: That doesn't mean you *need* to support source, if someone didn't provide source then they would get both. They could optionally use source to disambiguate.

**Comment by MJWeberg** [comment]
Finally getting back to this now. This is great, Will! Thanks you very much! I agree that linking in to the SQLite database would be nice, but is out of the scope of this PR. For now, people can still use the _eis_catalog_ GUI to query and download observations based on the metadata.

**Pull request merged**